packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -0,0 +1,362 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/gestures.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+import 'layout_builder.dart';
+import 'scroll_context.dart';
+import 'scroll_controller.dart';
+import 'scroll_physics.dart';
+import 'scroll_position.dart';
+import 'scroll_position_with_single_context.dart';
+import 'scroll_simulation.dart';
+
+/// The signature of a method that provides a [BuildContext] and
+/// [ScrollController] for building a widget that may overflow the draggable
+/// [Axis] of the containing [DraggableScrollSheet].
+///
+/// Users should apply the [scrollController] to a [ScrollView] subclass, such
+/// as a [SingleChildScrollView], [ListView] or [GridView], to have the whole
+/// sheet be draggable.
+typedef ScrollableWidgetBuilder = Widget Function(
+  BuildContext context,
+  ScrollController scrollController,
+);
+
+/// A widget that can be dragged along the vertical axis between its
+/// `minChildSize`, which defaults to `0.25` and `maxChildSize`, which defaults
+/// to `1.0`. These sizes are percentages of the height of the parent container.
+///
+/// The widget coordinates resizing and scrolling of the widget returned by
+/// `builder` as the user drags along the horizontal axis.
+///
+/// The widget will initially be displayed at its `initialChildSize`, which
+/// defaults to `0.5`. To enable dragging of the widget, ensure that the
+/// `builder` creates a widget that uses the provided [ScrollController]. If
+/// the widget created by the [ScrollableWidgetBuilder] does not use the
+/// provided [ScrollController], the sheet will remain at the
+/// `initialChildSize`.
+///
+/// {@tool sample}
+/// ```dart
```

Comment:
nit: There should be some prose text within the sample section explaining what this sample does.